### PR TITLE
feat: stepper: adds timeline variant

### PR DESCRIPTION
**Pull request by dkilgore-eightfold**
## SUMMARY:
Adds `Stepper` `timeline` variant
Fixes 2 A11y bugs by placing `aria-hidden="true"` on non `li` elements in `ul`
Fixes minor typo in `Stepper`
Adds `classNames` from props to `mergedClasses` of `Stepper`
Updates `Stepper` snaps and UTs

https://user-images.githubusercontent.com/99700808/212434404-ed8eb932-2bb6-45db-89e9-f3d22062203c.mp4

## JIRA TASK (Eightfold Employees Only):
ENG-39994

## CHANGE TYPE:

- [ ] Bugfix Pull Request
- [X] Feature Pull Request

## TEST COVERAGE:

- [ ] Tests for this change already exist
- [X] I have added unittests for this change

## TEST PLAN:
Pull the PR branch and run `yarn` and `yarn storybook`. Verify the `Stepper` and `Timeline` stories behave as expected.

### CODE REVIEW

File: src/components/Stepper/Stepper.tsx [hunk 1]
```diff
@@ -143,29 +140,63 @@ export const Stepper: FC<StepperProps> = React.forwardRef(
...
-      <div className={classes}>{index + 1}</div>
-    );
+    const circle = (
+      index: number,
+      classes: string,
+      complete: boolean,
+      size: StepSize,
+      active?: boolean,
+      icon?: IconName
+    ): JSX.Element => {
```
Comment (jhoward-eightfold) [inline src/components/Stepper/Stepper.tsx:178]: I think this JSX.Element is causing issues in typescript implementations where things are expecting React nodes or something along those lines. When i return to that effort I'll sync up with you on it. Just an FYI. no changes requested

File: src/components/Stepper/Stepper.tsx [hunk 2]
```diff
@@ -143,29 +140,63 @@ export const Stepper: FC<StepperProps> = React.forwardRef(
...
+        <div className={classes}>
+          <Icon
+            path={
+              complete || active
+                ? icon
+                  ? icon
+                  : IconName.mdiCircle
+                : icon
+                ? icon
+                : IconName.mdiCircleOutline
```
Comment (jhoward-eightfold) [inline src/components/Stepper/Stepper.tsx:192]: oh my. might be worth a method to improve readability

File: src/components/Stepper/Stepper.tsx [hunk 3]
```diff
@@ -359,28 +466,92 @@ export const Stepper: FC<StepperProps> = React.forwardRef(
                 >
                   <ul className={styles.steps} ref={stepsRef}>
                     {steps.map((step: Step, index: number) => {
-                      let stepItem = null;
+                      const combinedStepSize: StepSize = getCombinedStepSize(
```
Comment (jhoward-eightfold) [inline src/components/Stepper/Stepper.tsx:469]: what is a combinedStep? bit confused about this. Might be worth some code documentation

File: src/components/Stepper/Stepper.tsx [hunk 4]
```diff
@@ -359,28 +466,92 @@ export const Stepper: FC<StepperProps> = React.forwardRef(
...
+                      const mergedTheme: StepperThemeName = step.theme ?? theme;
+                      const mergedStatus: StepperValidationStatus =
+                        step.status ?? status;
+                      let stepItem: JSX.Element;
+                      let nodeAriaLabel: string;
+                      if (!step.nodeAriaLabelText) {
+                        nodeAriaLabel = locale!.lang.nodeAriaLabelText;
+                      } else {
+                        nodeAriaLabel = step.nodeAriaLabelText;
+                      }
```
Comment (jhoward-eightfold) [inline src/components/Stepper/Stepper.tsx:481]: could be a const with a ternary assignment

**Review by jhoward-eightfold — APPROVED** [review]
do you need to export the component anywhere or is it already covered by wildcard export of Stepper?

File: src/components/Stepper/Stepper.tsx (hunk 3 above)
Comment (dkilgore-eightfold) [reply src/components/Stepper/Stepper.tsx:469]: Each step has a unique set of props, this takes the aggregate and determines the overall spacing of the timeline.

**Pull request merged**